packages/devtools_app/lib/src/debugger/codeview.dart

```diff
@@ -808,12 +809,67 @@ class ScriptPopupMenuOption {
   }
 }
 
-final defaultScriptPopupMenuOptions = [copyScriptNameOption];
+final defaultScriptPopupMenuOptions = [copyScriptNameOption, goToLineOption];
 
 final copyScriptNameOption = ScriptPopupMenuOption(
   label: 'Copy filename',
   icon: Icons.content_copy,
-  onSelected: (scriptRef) => Clipboard.setData(
-    ClipboardData(text: scriptRef?.uri),
+  onSelected: (_, controller) => Clipboard.setData(
+    ClipboardData(text: controller.scriptLocation.value?.scriptRef?.uri),
   ),
 );
+
+void showGoToLineDialog(BuildContext context, DebuggerController controller) {
+  showDialog(
+    context: context,
+    builder: (context) => GoToLineDialog(controller),
+  );
+}
+
+const goToLineOption = ScriptPopupMenuOption(
+  label: 'Go to line number',
+  icon: Icons.list,
+  onSelected: showGoToLineDialog,
+);
+
+class GoToLineDialog extends StatelessWidget {
+  const GoToLineDialog(this._debuggerController);
+
+  final DebuggerController _debuggerController;
+
+  @override
+  Widget build(BuildContext context) {
+    return DevToolsDialog(
+      title: dialogTitleText(Theme.of(context), 'Go To'),
+      content: Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          TextField(
+            autofocus: true,
+            onSubmitted: (value) {
+              if (value.isNotEmpty) {
+                Navigator.of(context, rootNavigator: true).pop('dialog');
+                final line = int.parse(value);
```

Comment:
we should fail gracefully if int.parse fails or if the value is a negative number.

Reply:
It can't be a negative number due to the limited input.